### ISSUE TITLE
False positive on DeprecatedCall for WebView constructor

### ISSUE THREAD

**Issue by edenman** [slackhq/slack-lints#268]
As far as I can tell, WebView is not actually deprecated, but slack-lint is claiming that it is.  Maybe something wrong with the annotation detector?

```
Lint found 6 errors, 0 warnings (10 errors filtered by baseline lint-baseline.xml). First failure:

/Users/edenman/workspace/doggifs/src/main/java/org/coffeetrain/doggifs/SetAvailableGifsScreenView.kt:23: Error: android.webkit.WebView.WebView is deprecated; consider using an alternative. [DeprecatedCall from slack-lint]
    val foo = WebView(context).apply {
```

Repro project: https://github.com/edenman/doggifs/compare/edenman/webview-deprecated?expand=1

**Comment by edenman** [comment]
Maybe it's confused because there is a different constructor that is deprecated?
```
@Deprecated
    public WebView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr,
            boolean privateBrowsing)
```

**Comment by ZacSweers** [comment]
ah, thanks for the report. Maybe we're not resolving the correct constructor from the callsite

**Comment by ZacSweers** [comment]
I think this is a bug in lint, we're just using their `AbstractAnnotationDetector()` and it's handing us these symbols. It seems it's resolving the wrong one when resolving overloads. Do you see it in Lint 8.7+?

**Comment by ZacSweers** [comment]
Instructions for the above ^^ 

https://googlesamples.github.io/android-custom-lint-rules/usage/newer-lint.md.html

**Comment by kozaxinan** [comment]
I updated repro repo with new agp. https://github.com/ZeeshanShabbir/lint-investigation
the issue still exists on new agp and lint.

**Comment by ZacSweers** [comment]
Could you file an issue on the Android lint issue tracker? At this point it's not really something we can control, it appears lint is reporting the wrong overload

**Comment by kozaxinan** [comment]
I reported to Google. https://issuetracker.google.com/issues/371686443

**Comment by ZacSweers** [comment]
Marked as fixed upstream

> Android Studio Narwhal Feature Drop | 2025.1.2 Canary 5
> Android Gradle Plugin 8.12.0-alpha05
